service/asset_service.go

```diff
@@ -21,6 +25,35 @@ func NewAssetService(repo repository.AssetRepository) AssetService {
 		assetRepo: repo,
 	}
 }
+func (service *assetService) DeleteAsset(ctx context.Context, Id uuid.UUID) (*domain.Asset, error) {
+	asset, err := service.assetRepo.DeleteAsset(ctx, Id)
+	if err == customerrors.AssetAlreadyDeleted {
+		return nil, customerrors.AssetAlreadyDeleted
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if asset == nil {
+		return nil, customerrors.NoAssetsExist
+	}
+	return asset, nil
+}
+
+func (service *assetService) UpdateAsset(ctx context.Context, Id uuid.UUID, req contract.UpdateRequest) (*domain.Asset, error) {
+	asset, err := service.assetRepo.UpdateAsset(ctx, Id, req)
+	if err == customerrors.AssetAlreadyDeleted {
+		return nil, customerrors.AssetAlreadyDeleted
+	}
+	if err != nil {
```

Comment:
same as above

Reply:
actually in update and delete i am checking two condition apart from nill asset already deleted if it is already deleted then no updation can be performed that why is condition and if there is no asset exist then no updation can be performed